frontend/src/routes/ManageAdmissions/components/GroupSelector.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from "react";
+import { Group } from "src/types";
+import { callApiFromQuery } from "src/utils/callApi";
+import styled from "styled-components";
+
+interface GroupSelectorProps {
+  value: Group[];
+  toggleGroup: (group: Group) => void;
+}
+
+const GroupSelector: React.FC<GroupSelectorProps> = ({
+  value: selectedGroups,
+  toggleGroup,
+}) => {
+  const [groups, setGroups] = useState<Group[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      const res = await callApiFromQuery("/group/");
+      setGroups(res);
+    })();
```

Comment:
Please make the outer function async

Reply:
The reason I added the inner function in the first place is that useEffect doesn't allow async functions as the main function, so I had to add that inner function to make it work